openapi.yaml

```diff
@@ -17,1237 +21,222 @@ produces:
-  NewPhoneResultObject:
-    properties:
-      data:
-        $ref: '#/definitions/NewPhoneResourceObject'
-  NewPhoneResourceObject:
-    properties:
-      type:
-        $ref: '#/definitions/PhoneType'
-      attributes:
-        required:
-          - addressType
-          - phoneType
-          - areaCode
-          - phoneNumber
-          - primaryIndicator
-        properties: &commonPhoneAttributes
-          areaCode:
-            type: string
-            description: Area code for the phone number.
-            example: '541'
-            pattern: '^[0-9]{1,3}$'
-          phoneNumber:
-            type: string
-            description: Phone number not including the area code.
-            example: '4443333'
-            pattern: '^[0-9]{1,7}$'
-          phoneExtension:
-            type: string
-            description: Extension for the phone number. Can be returned in different formats.
-            example: '20'
-            pattern: '^[0-9]{1,4}$'
-          primaryIndicator:
-            type: boolean
-            description: Indicates if this phone number is marked as the primary number.
-            example: true
-          phoneType:
-            type: string
-            description: Enum marking which type of phone number this is.
-            example: CM
-            enum: *PHONETYPEENUM
-          addressType:
-            type: string
-            description: Enum marking which type of address this phone number matches with.
-            example: CM
-            enum: *ADDRESSTYPEENUM
-  PhonesResultObject:
-    properties:
-      links:
-        $ref: '#/definitions/SelfLink'
-      data:
-        type: array
-        items:
-          $ref: '#/definitions/PhoneResourceObject'
-  PhoneResultObject:
-    properties:
-      links:
-        $ref: '#/definitions/SelfLink'
-      data:
-        $ref: '#/definitions/PhoneResourceObject'
-  PhoneResourceObject:
-    properties:
-      id:
-        type: string
-        description: Globally unique ID for the phones.
-      type:
-        $ref: '#/definitions/PhoneType'
-      attributes:
-        properties:
-          <<: *commonPhoneAttributes
-          fullPhoneNumber:
-            type: string
-            description: Combination of area code and phone number.
-            example: '5414443333'
-          phoneTypeDescription:
-            type: string
-            description: Describes the phone type enum.
-            example: Current
-          addressTypeDescription:
-            type: string
-            description: Describes the address type enum.
-            example: Current Mailing
-          lastModified:
-            type: string
-            description: The date of the most recent change made to this record.
-            example: '2015-10-19'
+              firstName: *firstName
+              lastName: *lastName
+              middleName: *middleName
+              birthDay: *birthDay
+              citizen: *citizen
+              sex: *sex
```

Comment:
Instead of referring each filed here, what do you think if you define a shared object called `commonPersonObject` or something else?

Reply:
Yeah I thought about that but I really like having all the attributes defined in one place instead of spreading it out over multiple objects. Might be good to bring up in our next meeting.

Reply:
Hmm, not sure did I misunderstand, but I mean you can still define them in one place and use `allOf` to merge when needed: https://github.com/osu-mist/express-api-skeleton/blob/86a6439fac923b34fbd936796fd3d161ba49eadc/openapi.yaml#L209-L211. So it will be something like:

```yaml
Resource:
    - $ref: '#/definitions/CommonPart'
    - properties:
        extraField1:
          type: string
        extraField2:
          type: string
```

Reply:
In my code all of the person fields are defined under the `PersonResource` while in your example some are defined in `CommonPart` with some extra fields under `Resource`.  I just find it easier when I have to edit these fields in the future if they are all found in one place.